### PR TITLE
Ufal/synchronising cmdi outputs

### PR DESCRIPTION
**Pull request by MajoBerger**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |
## Analysis
(Write here, if there is needed describe some specific problem. Erase it, when it is not needed.)
## Problems
(Write here, if some unexpected problems occur during solving issues. Erase it, when it is not needed.)

### CODE REVIEW

File: dspace-oai/src/main/java/org/dspace/utils/LicenseUtil.java [hunk 1]
```diff
@@ -214,14 +214,16 @@ public static org.w3c.dom.NodeList uriToRestrictions(String uri)
...
-            Element res = doc.createElement("restriction");
+            Element res = doc.createElement("restriction" + x);
+            x++;
             res.appendChild(doc.createTextNode(restriction));
             root.appendChild(res);
         }
 
-        return root.getElementsByTagName("restriction");
+        return root.getChildNodes();
+//        return root.getElementsByTagName("restriction");
```
Comment (vidiecan) [inline dspace-oai/src/main/java/org/dspace/utils/LicenseUtil.java:226]: why //? remove if incorrect

File: dspace-oai/src/main/java/org/dspace/utils/SpecialItemService.java [hunk 2]
```diff
@@ -129,16 +129,27 @@ public static Node getFunding(String mdValue) {
             }
             String[] values = mdValue
                     .split(DCInput.ComplexDefinitions.getSeparator(), -1);
-            // mind the order in input forms, org;code;projname;type
-            Element[] elements = {organization, code, projName, fundsType};
+            // mind the ORIGINAL order in input forms, org;code;projname;type
+            // however, the order in the XML is different due to reasons unknown to me
```
Comment (vidiecan) [inline dspace-oai/src/main/java/org/dspace/utils/SpecialItemService.java:133]: better docs,

File: dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/NodeListXslFunction.java [hunk 3]
```diff
@@ -75,11 +78,34 @@ final public XdmValue call(XdmValue[] xdmValues) throws SaxonApiException {
 
 
         NodeList nodeList = getNodeList(val);
-        Node oneNode = nodeList.item(0);
+//        Node oneNode = nodeList.item(0);
```
Comment (vidiecan) [inline dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/NodeListXslFunction.java:81]: ?

File: dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java [hunk 4]
```diff
@@ -290,6 +297,24 @@ public static Metadata retrieveMetadata(Context context, Item item) {
...
+//                if(ld.getRequiredInfo() != null && ld.getRequiredInfo().length() > 0){
+//                    restricted = true;
+//                }
+//                if(restricted){
+//                    break;
+//                }
+//            }
+//        }
+
+        if (restricted) {
```
Comment (vidiecan) [inline dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java:313]: ?

File: dspace/config/clarin-dspace.cfg [hunk 5]
```diff
@@ -149,7 +149,7 @@ disable.ssl.check.specific.requests = false
 
 ##### Matomo statistics #####
 # Auth token
-matomo.track.enabled = true
+matomo.track.enabled = false
```
Comment (vidiecan) [inline dspace/config/clarin-dspace.cfg:152]: ?

File: dspace/config/crosswalks/oai/metadataFormats/metasharev2.xsl [hunk 6]
```diff
@@ -285,10 +285,10 @@
                                 <xsl:for-each select="doc:metadata/doc:element[@name='local']/doc:element[@name='sponsor']/doc:element/doc:field[@name='value']">
                                                 <ms:fundingProject>
                                                     <ms:projectName>
-                                                            <xsl:value-of select="tokenize(., ';')[3]"/>
+                                                            <xsl:value-of select="tokenize(., ';')[4]"/>
```
Comment (vidiecan) [inline dspace/config/crosswalks/oai/metadataFormats/metasharev2.xsl:288]: docs?

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-oai/src/main/java/org/dspace/utils/SpecialItemService.java (hunk 2 above)
Comment (MajoBerger) [reply dspace-oai/src/main/java/org/dspace/utils/SpecialItemService.java:133]: comment about changed order

File: dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/NodeListXslFunction.java (hunk 3 above)
Comment (MajoBerger) [reply dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/NodeListXslFunction.java:81]: add todo, needs finishing

File: dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java (hunk 4 above)
Comment (MajoBerger) [reply dspace-oai/src/main/java/org/dspace/xoai/util/ItemUtils.java:313]: add to issue

File: dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/NodeListXslFunction.java (hunk 3 above)
Comment (MajoBerger) [reply dspace-oai/src/main/java/org/dspace/xoai/services/impl/resources/functions/NodeListXslFunction.java:81]: add also followup issue

File: dspace/config/clarin-dspace.cfg (hunk 5 above)
Comment (MajoBerger) [reply dspace/config/clarin-dspace.cfg:152]: shoudl not be here

File: dspace/config/crosswalks/oai/metadataFormats/metasharev2.xsl (hunk 6 above)
Comment (MajoBerger) [reply dspace/config/crosswalks/oai/metadataFormats/metasharev2.xsl:288]: also add comment (see file from up there)

**Review by vidiecan — DISMISSED** [review]

**Review by vidiecan — APPROVED** [review]

**Review by milanmajchrak — APPROVED** [review]

**Pull request merged**